todo_app/data/session_items.py

```diff
@@ -43,9 +53,26 @@ def add_item(title):
     items = get_items()
 
     # Determine the ID for the item based on that of the previously added item
-    id = items[-1]['id'] + 1 if items else 0
-
-    item = { 'id': id, 'title': title, 'status': 'Not Started' }
+    #id = items[-1]['id'] + 1 if items else 0
+    id = 0
+    checkid = []
+    for item in items:
+        checkid.append(item['id'])
+    
+    if items != []:
+        matchfound = True
+        while matchfound:
+            if (id in checkid):
+                #print(str(id) + " found!")
+                id = id + 1
+            else:
+                matchfound = False
```

Comment:
This is overcomplicated just to try and re-use ids and I don't see the benefit. You could replace this block with just `id = max(checkid) + 1`

In fact, you would more often want to *avoid* re-using deleted items' ids in case there's a problem when someone comes looking for a deleted item and you give them an unrelated one. But that would require keeping a record of the latest id.